### PR TITLE
Consolidate Asset Store and HCMS docs

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Unify Asset Store and Headless CMS docs into one Rafter component
- Remove Asset Store and Headless CMS components from docs
- Rename DocTopic into AssetGroup
- Update labels
- Describe new AssetGroup functionality (adding BucketRef to AssetGroup CR definition)
- Review the existing docs and remove any outdated info
- Consolidate 4 tutorials for switching into Gateway mode into one with toggles for different cloud providers

TO DO (after Rafter is introduced in Kyma):
- Verify values in conf docs
- Verify names of custom metrics
- Update the DocsTopic fo Rafter into AssetGroup + change the component doc order

**Related issue(s)**
See also #https://github.com/kyma-project/rafter/issues/10

### CODE REVIEW

File: docs/rafter/01-02-rafter.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Overview
+type: Overview
+---
+
+In general, Rafter is a new breed of traditional Content Management Systems (CMS) that provides a way of storing and managing raw content, and exposing it through an API. It allows you to pull the content into your own application and tailor it to your needs, delivering it in any format, on any device. Contrary to the traditional CMS, such as WordPress, the Headless CMS does not provide a display layer and ready-to-use templates. Instead, it only ensures a database backend. It gives flexibility on the choice of the frontend, thus cutting the default "head" off the traditional CMS solutions.
```
Comment (michal-hudy) [inline docs/rafter/01-02-rafter.md]:
```suggestion
In general, Rafter is a new breed of traditional Content Management Systems (CMS) that provides a way of storing and managing raw content, and exposing it through an API. It allows you to pull the content into your own application and tailor it to your needs, delivering it in any format, on any device. Contrary to the traditional CMS, such as WordPress, Rafter does not provide a display layer and ready-to-use templates. Instead, it only ensures a database backend. It gives flexibility on the choice of the frontend, thus cutting the default "head" off the traditional CMS solutions.
```

File: docs/rafter/02-01-basic-flow.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Resources
+
+The whole concept of Rafter relies on the following components:
+
+- **Asset custom resource** (CR) is an obligatory [CR](#custom-resource-asset) in which you define the asset you want to store in a given storage bucket. Its definition requires the asset name and mode, the name of the Namespace in which it is available, the address of its web location, and the name of the bucket in which you want to store it. Optionally, you can specify the validation and mutation requirements that the asset must meet before it is stored.
+
+- **Asset Controller** (AC) manages the [Asset CR lifecycle](#details-asset-custom-resource-lifecycle).
+
+- **AssetGroup custom resource** (CR) orchestrates the creation of multiple Asset CRs for a specific documentation topic in a given Namespace.
```
Comment (michal-hudy) [inline docs/rafter/02-01-basic-flow.md]: Maybe without mentioning about documentation
```suggestion
- **AssetGroup custom resource** (CR) orchestrates the creation of multiple Asset CRs in a given Namespace.
```

File: docs/rafter/02-01-basic-flow.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+- **AssetGroup Controller** creates Asset custom resources (CRs) based on an AssetGroup CR definition. If the AssetGroup CR defines two sources of documentation topics, such as `asyncapi` and `markdown`, the AssetGroup Controller creates two Asset CRs. The AssetGroup Controller also monitors the status of the Asset CR and updates the status of the AssetGroup CR accordingly.
```
Comment (michal-hudy) [inline docs/rafter/02-01-basic-flow.md]:
```suggestion
- **AssetGroup Controller** creates Asset custom resources (CRs) based on an AssetGroup CR definition. If the AssetGroup CR defines two sources of assets, such as `asyncapi` and `markdown`, the AssetGroup Controller creates two Asset CRs. The AssetGroup Controller also monitors the status of the Asset CR and updates the status of the AssetGroup CR accordingly.
```

File: docs/rafter/02-01-basic-flow.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- **Bucket CR** is an obligatory [CR](#custom-resource-bucket) in which you define the name of the bucket for storing assets.
+
+- **Bucket Controller** manages the [Bucket CR lifecycle](#details-bucket-custom-resource-lifecycle).
+
+- **Validation Service** is an optional service which ensures that the asset meets the validation requirements specified in the Asset CR before uploading it to the bucket. The service returns the validation status to the AC.
+
+- **Mutation Service** is an optional service which ensures that the asset is modified according to the mutation specification defined in the Asset CR before it is uploaded to the bucket. The service returns the modified asset to the AC.
+
+- [**Front Matter Service**](#details-front-matter-service) is an optional service which extracts metadata from assets. The metadata information is stored in the CR status. The service returns the asset metadata to the AC.
```
Comment (michal-hudy) [inline docs/rafter/02-01-basic-flow.md]: Why specific service is mentioned for extractor, when for validation/mutation is generic?

File: docs/rafter/02-01-basic-flow.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+>**NOTE:** This flow also applies to the cluster-wide counterparts of all CRs.
+
+![](./assets/basic-architecture.svg)
+
+1. Create an AssetGroup CR that manages a single asset or a package of assets you want to upload within a specified category.
+2. Rafter creates Asset CRs in the number specified in the AssetGroup CR.
+3. Services implemented for Rafter webhooks, can optionally validate, mutate, or extract data from assets before uploading them into buckets.
+4. Rafter creates Bucket CRs that define buckets in which assets must be stored.
+5. Rafter creates buckets in Minio Gateway and moves assets into them.
```
Comment (michal-hudy) [inline docs/rafter/02-01-basic-flow.md]: In fact, the order is slightly different 😄 
1. Create an AssetGroup...
2. Rafter creates Bucket CRs if not set in AssetGroup
3. Rafter creates Assets CRs ....

File: docs/rafter/02-01-basic-flow.md (hunk 4 above)
Comment (kazydek) [reply docs/rafter/02-01-basic-flow.md]: Good question. I would need to think if that is actually required here where we mention generic architecture and not specific webhook implementations.

File: docs/rafter/02-01-basic-flow.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+>**NOTE:** All CRs and controllers have their cluster-wide counterparts, names of which start with the **Cluster** prefix, such as ClusterAssetGroup CR.
+
+## Basic content flow
+
+This diagram shows a high-level overview of how Rafter works:
+
+>**NOTE:** This flow also applies to the cluster-wide counterparts of all CRs.
+
+![](./assets/basic-architecture.svg)
```
[the remark this answers is not in the document]
Comment (derberg) [reply docs/rafter/02-01-basic-flow.md]: try removing AssetGroup from it and simplify it more

File: docs/rafter/01-02-rafter.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+- [Upload Service](#details-upload-service) (optional service)
+- [AsyncAPI Service](#details-asyncapi-service) (extension)
+- [Front Matter Service](#details-front-matter-service) (extension)
+
+This solution offers a number of benefits:
+
+- It is flexible. You can use it for storing various types of assets, such as Markdown documents, ZIP, PNG, or JS files.
+- It is scalable. It allows you to store assets on a production system, using cloud provider storage services. At the same time, you can apply it to local development and use MinIO to store assets on-premise.
+- It is multi-cloud and not locked into one vendor. When using the Asset Store in a production system, you can seamlessly switch between different major service providers, such as AWS S3 or Azure Blob.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/01-02-rafter.md]:
```
- It is not locked into one vendor. When using Rafter in a production system, you can seamlessly switch between different major service providers, such as AWS S3 or Azure Blob.

```
?

File: docs/rafter/02-01-basic-flow.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+---
+title: Basic content flow
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-01-basic-flow.md]: Ok, I changed the title into `Basic asset flow` and named section only as `Flow`.

File: docs/rafter/02-01-basic-flow.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+type: Architecture
+---
+
+## Resources
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-01-basic-flow.md]: Hmm, it is a bit strange order for me as I guess you would first like to know the terms and "main actors" and then get to know the flow but I can change this order if you wish.

File: docs/rafter/01-02-rafter.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (derberg) [reply docs/rafter/01-02-rafter.md]: `When using Rafter you avoid vendor lock-in`

I want this phrase here `vendor lock-in`

File: docs/rafter/02-01-basic-asset-flow.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+>**NOTE:** This flow also applies to the cluster-wide counterparts of all CRs.
+
+![](./assets/basic-architecture.svg)
+
+1. Create a Bucket CR. This propagates the creation of buckets in MinIO Gateway where your assets will be stored.
+2. Create an Asset CR that contains the reference to your assets.
+3. Services implemented for Rafter webhooks can optionally validate, mutate, or extract data from assets before uploading them into buckets.
+4. Rafter uploads assets onto buckets in MinIO Gateway.
+
+Read more about the role of the main Rafter components and controllers that manage them:
```
Comment (bszwarc) [inline docs/rafter/02-01-basic-asset-flow.md]:
```suggestion
Read more about the role of main Rafter components and controllers that manage them:
```

File: docs/rafter/03-04-minio-gateway.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+You can also set MinIO credentials directly using `values.yaml` files. For more details, see the official [MinIO documentation](https://github.com/helm/charts/tree/master/stable/minio#configuration).
+
+
+## Production storage
+
+For the production purposes, Rafter uses MinIO Gateway which:
+
+- Is a multi-cloud solution that offers the flexibility to choose a given cloud provider for the specific Kyma installation, including Azure, Amazon, and Google.
+- Allows you to use various cloud providers that support the data replication and CDN configuration
```
Comment (bszwarc) [inline docs/rafter/03-04-minio-gateway.md]:
```suggestion
- Allows you to use various cloud providers that support the data replication and CDN configuration.
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+| **spec.sources.url** | Yes  | Specifies the location of a single file or a package. |
+| **spec.sources.filter** | No | Specifies a set of assets from the package to upload. The regex used in the filter must be [RE2](https://golang.org/s/re2syntax)-compliant. |
+| **status.lastheartbeattime** | Not applicable | Provides the last time when the ClusterAssetGroup Controller processed the ClusterAssetGroup CR. |
+| **status.message** | Not applicable | Describes a human-readable message on the CR processing progress, success, or failure. |
+| **status.phase** | Not applicable | The ClusterAssetGroup Controller adds it to the ClusterAssetGroup CR. It describes the status of processing the ClusterAssetGroup CR by the ClusterAssetGroup Controller. It can be `Ready`, `Pending`, or `Failed`. |
+| **status.reason** | Not applicable | Provides the reason why the ClusterAssetGroup CR processing succeeded, is pending, or failed. See the [**Reasons**](#status-reasons) section for the full list of possible status reasons and their descriptions.  |
+
+> **NOTE:** The ClusterAssetGroup Controller automatically adds all parameters marked as **Not applicable** to the ClusterAssetGroup CR.
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
>**NOTE:** The ClusterAssetGroup Controller automatically adds all parameters marked as **Not applicable** to the ClusterAssetGroup CR.
```

File: docs/rafter/03-05-supported-webhooks.md [hunk 13]
```diff
@@ -15,9 +15,9 @@ The Asset Store supports the following types of webhooks:
 
 ## Service specification requirements
 
-If you create a specific mutation, validation, or metadata service for the available webhooks and you want the Asset Store to properly communicate with it, you must ensure that the API exposed by the given service meets the API contract requirements. These criteria differ depending on the webhook type:
+If you create a specific mutation, validation, or metadata service for the available webhooks and you want Rafter to properly communicate with it, you must ensure that the API exposed by the given service meets the API contract requirements. These criteria differ depending on the webhook type:
 
-> **NOTE:** Services are described in the order in which the Asset Store processes them.
+> **NOTE:** Services are described in the order in which Rafter processes them.
```
Comment (bszwarc) [inline docs/rafter/03-05-supported-webhooks.md]:
```suggestion
>**NOTE:** Services are described in the order in which Rafter processes them.
```

File: docs/rafter/06-02-assetgroup.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+| **spec.sources.url** | Yes | Specifies the location of a single file or a package. |
+| **spec.sources.filter** | No | Specifies a set of assets from the package to upload. The regex used in the filter must be [RE2](https://golang.org/s/re2syntax)-compliant. |
+| **status.lastheartbeattime** | Not applicable | Provides the last time when the AssetGroup Controller processed the AssetGroup CR. |
+| **status.message** | Not applicable | Describes a human-readable message on the CR processing progress, success, or failure. |
+| **status.phase** | Not applicable | The AssetGroup Controller adds it to the AssetGroup CR. It describes the status of processing the AssetGroup CR by the AssetGroup Controller. It can be `Ready`, `Pending`, or `Failed`. |
+| **status.reason** | Not applicable | Provides the reason why the AssetGroup CR processing succeeded, is pending, or failed. See the [**Reasons**](#status-reasons) section for the full list of possible status reasons and their descriptions. |
+
+> **NOTE:** The AssetGroup Controller automatically adds all parameters marked as **Not applicable** to the AssetGroup CR.
```
Comment (bszwarc) [inline docs/rafter/06-02-assetgroup.md]:
```suggestion
>**NOTE:** The AssetGroup Controller automatically adds all parameters marked as **Not applicable** to the AssetGroup CR.
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+
+### Status reasons
+
+Processing of a ClusterAssetGroup CR can succeed, continue, or fail for one of these reasons:
+
+| Reason | Phase | Description |
+| --------- | ------------- | ----------- |
+| `AssetCreated` | `Pending` | The ClusterAssetGroup Controller created the specified asset. |
+| `AssetCreationFailed` | `Failed` | The ClusterAssetGroup Controller couldn't create the specified asset due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetCreationFailed` | `Failed` | The ClusterAssetGroup Controller couldn't create the specified asset due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetsCreationFailed` | `Failed` | The ClusterAssetGroup Controller couldn't create assets due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetsCreationFailed` | `Failed` | The ClusterAssetGroup Controller couldn't create assets due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetsListingFailed` | `Failed` | The ClusterAssetGroup Controller couldn't list assets due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetsListingFailed` | `Failed` | The ClusterAssetGroup Controller couldn't list assets due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetDeleted` | `Pending` | The ClusterAssetGroup Controller deleted specified assets. |
+| `AssetDeletionFailed` | `Failed`  | The ClusterAssetGroup Controller couldn't delete the specified asset due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetDeletionFailed` | `Failed`  | The ClusterAssetGroup Controller couldn't delete the specified asset due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetsDeletionFailed` | `Failed` | The ClusterAssetGroup Controller couldn't delete assets due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetsDeletionFailed` | `Failed` | The ClusterAssetGroup Controller couldn't delete assets due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetUpdated` | `Pending` | The ClusterAssetGroup Controller updated the specified asset. |
+| `AssetUpdateFailed` | `Failed` | The ClusterAssetGroup Controller couldn't upload the specified asset due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetUpdateFailed` | `Failed` | The ClusterAssetGroup Controller couldn't upload the specified asset due to an error. |
```

File: docs/rafter/06-05-clusterassetgroup.md [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+| `AssetsUpdateFailed` | `Failed` | The ClusterAssetGroup Controller couldn't update assets due to the provided error. |
```
Comment (bszwarc) [inline docs/rafter/06-05-clusterassetgroup.md]:
```suggestion
| `AssetsUpdateFailed` | `Failed` | The ClusterAssetGroup Controller couldn't update assets due to an error. |
```

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 22]
```diff
@@ -0,0 +1,391 @@
...
+- [Alibaba Cloud](https://alibabacloud.com) account
+
+  </details>
+
+</div>
+
+
+## Steps
+
+You can set MinIO to the given Gateway mode both during and after Kyma installation. In both cases, you need to create and configure an access key for your cloud provider account, apply a Secret and a ConfigMap with an override onto a cluster or Minikube, and trigger the Kyma installation process.
```
Comment (bszwarc) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]:
```suggestion
You can set MinIO to the given Gateway mode both during and after Kyma installation. In both cases, you need to create and configure an access key for your cloud provider account, apply a Secret and a ConfigMap with an override to a cluster or Minikube, and trigger the Kyma installation process.
```

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 23]
```diff
@@ -0,0 +1,391 @@
...
+
+  </details>
+  <details>
+  <summary label="aws-s3">
+  AWS S3
+  </summary>
+
+Create an AWS access key for an IAM user. Follow these steps:
+
+1. Access the [AWS Identity and Access Management console](https://console.aws.amazon.com/iam/)
```
Comment (bszwarc) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]:
```suggestion
1. Access the [AWS Identity and Access Management console](https://console.aws.amazon.com/iam/).
```

File: docs/rafter/08-02-set-minio-to-gateway-mode.md [hunk 24]
```diff
@@ -0,0 +1,391 @@
...
+      export ALIBABA_SECRET_KEY={your-secret-key}
+      ```
+
+  </details>
+
+</div>
+
+### Configure MinIO Gateway mode
+
+Apply the following Secret and ConfigMap with an override onto a cluster or Minikube. Run:
```
Comment (bszwarc) [inline docs/rafter/08-02-set-minio-to-gateway-mode.md]:
```suggestion
Apply the following Secret and ConfigMap with an override to a cluster or Minikube. Run:
```

File: docs/rafter/11-02-asyncapi-service.md [hunk 25]
```diff
@@ -0,0 +1,31 @@
+---
+title: AsyncAPI Service
+type: Metrics
+---
+
+This table shows the AsyncAPI Service custom metrics, their types, and descriptions.
+
+| Name | Type | Description |
+|------|-------------|------|
+| `rafter_services_http_request_and_mutation_duration_seconds` | histogram | Specifies a number of assets that the service received for processing and mutated within a given time series. |
```
Comment (bszwarc) [inline docs/rafter/11-02-asyncapi-service.md]: Should it be `a` number or `the` number in this case and in the following rows? I think it should be `the number` as we point to the specific number of assets. 🤔

File: docs/rafter/11-02-asyncapi-service.md [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+kubectl -n kyma-system port-forward svc/rafter-asyncapi-service 80
+```
+
+To check the metrics, open a new terminal window and run:
+
+```bash
+curl http://localhost:80/metrics
+```
+
+> **TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `80`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-asyncapi-service 8080:80` and update the port in the localhost address.
```
Comment (bszwarc) [inline docs/rafter/11-02-asyncapi-service.md]:
```suggestion
>**TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `80`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-asyncapi-service 8080:80` and update the port in the localhost address.
```

File: docs/rafter/11-03-upload-service.md [hunk 27]
```diff
@@ -0,0 +1,29 @@
+---
+title: Upload Service
+type: Metrics
+---
+
+This table shows the Upload Service custom metrics, their types, and descriptions.
+
+| Name | Type | Decription |
+|------|-------------|------|
+| `rafter_upload_service_http_request_duration_seconds` | histogram | Specifies a number of HTTP requests the service processes in a given time series. |
```
Comment (bszwarc) [inline docs/rafter/11-03-upload-service.md]: Also the question of `a number` vs `the number`

File: docs/rafter/11-03-upload-service.md [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+kubectl -n kyma-system port-forward svc/rafter-upload-service 3000
+```
+
+To check the metrics, open a new terminal window and run:
+
+```bash
+curl http://localhost:3000/metrics
+```
+
+> **TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `3000`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-upload-service 8080:3000` and update the port in the localhost address.
```
Comment (bszwarc) [inline docs/rafter/11-03-upload-service.md]:
```suggestion
>**TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `3000`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-upload-service 8080:3000` and update the port in the localhost address.
```

File: docs/rafter/11-04-front-matter-service.md [hunk 29]
```diff
@@ -1,21 +1,21 @@
...
 ---
 
-This table shows the Asset Metadata Service custom metrics, their types, and descriptions.
+This table shows the Front Matter Service custom metrics, their types, and descriptions.
 
 | Name | Type | Description |
 |------|-------------|------|
-| `assetstore_metadata_service_http_request_duration_seconds` | histogram | Specifies a number of HTTP requests the service processes in a given time series. |
-| `assetstore_metadata_service_http_request_returned_status_code` | counter | Specifies a number of different HTTP response status codes in a given time series. |
+| `rafter_front_matter_service_http_request_duration_seconds` | histogram | Specifies a number of HTTP requests the service processes in a given time series. |
```
Comment (bszwarc) [inline docs/rafter/11-04-front-matter-service.md]: ` a` and `the` number :)

File: docs/rafter/11-04-front-matter-service.md [hunk 30]
```diff
@@ -24,6 +24,6 @@ To check the metrics, open a new terminal window and run:
 curl http://localhost:80/metrics
 ```
 
-> **TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `80`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/assetstore-asset-metadata-service 8080:80` and update the port in the localhost address.
+> **TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `80`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-front-matter-service 8080:80` and update the port in the localhost address.
```
Comment (bszwarc) [inline docs/rafter/11-04-front-matter-service.md]:
```suggestion
>**TIP:** To use these commands, you must have a running Kyma cluster and kubectl installed. If you cannot access port `80`, redirect the metrics to another one. For example, run `kubectl -n kyma-system port-forward svc/rafter-front-matter-service 8080:80` and update the port in the localhost address.
```

File: docs/rafter/01-01-rafter.md [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+type: Overview
+---
+
+Technically speaking, Rafter is a solution for storing and managing different types of assets, such as documents, files, images, API specifications, and client-side applications. It uses an external solution, [MinIO](https://min.io/), for storing assets. The whole concept relies on Kubernetes custom resources (CRs) managed by Asset, Bucket, AssetGroup controllers (and their cluster-wide counterparts) grouped under the [Rafter Controller Manager](https://github.com/kyma-project/rafter/blob/master/cmd/manager/README.md). These CRs include:
+
+- Asset CR which manages a single asset or a package of assets
+- Bucket CR which manages buckets in which these assets are stored
+- AssetGroup CR which manages a group of Asset CRs of a specific type
+
+Rafter enables you to manage assets using supported webhooks. For example, if you use Rafter to store a file such as a specification, you can additionally define a webhook service that Rafter should call before the file is sent to storage. The webhook service can:
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/01-01-rafter.md]: Controler does it. Hmm, I will think if I will change or leave it as it is.

File: docs/rafter/02-01-basic-asset-flow.md [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+This diagram shows a high-level overview of how Rafter works:
+
+>**NOTE:** This flow also applies to the cluster-wide counterparts of all CRs.
+
+![](./assets/basic-architecture.svg)
+
+1. Create a Bucket CR. This propagates the creation of buckets in MinIO Gateway where your assets will be stored.
+2. Create an Asset CR that contains the reference to your assets.
+3. Services implemented for Rafter webhooks can optionally validate, mutate, or extract data from assets before uploading them into buckets.
+4. Rafter uploads assets onto buckets in MinIO Gateway.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-01-basic-asset-flow.md]: changed into `into` ;)

File: docs/rafter/02-02-assetgroup.md [hunk 33]
```diff
@@ -0,0 +1,21 @@
+---
+title: AssetGroup CR flow
+type: Architecture
+---
+
+This diagram gives more details of the AssetStore CR flow, controller that manages it, and processes that take place underneath:
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-02-assetgroup.md]: and `AssetStore CR ` into `AssetGroup CR `

File: docs/rafter/02-03-asset-and-bucket.md [hunk 34]
```diff
@@ -0,0 +1,20 @@
+---
+title: Asset and Bucket CRs flow
+type: Architecture
+---
+
+
+This diagram provides an overview of the basic Asset flow and the role of particular components in this process:
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-03-asset-and-bucket.md]: O changed it into `basic Asset and Bucket CRs flow `

File: docs/rafter/02-02-assetgroup.md [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+
+![](./assets/rafter-architecture-1.svg)
+
+1. Create an AssetGroup CR in a given Namespace.
+2. The AssetGroup Controller reads the AssetGroup CR definition.
+3. If the AssetGroup CR definition does not provide a reference name (**bucketRef**) of the Bucket CR, the AssetsGroup Controller checks if the default Bucket CR already exists in this Namespace. If it does not exist yet, the AssetsGroup Controller creates a new Bucket CR with:
+
+- the `rafter-public-{ID}` name, where `{ID}` is a randomly generated string, such as `rafter-public-6n32wwj5vzq1k`.
+- the **rafter.kyma-project.io/default: true** label
+- the **rafter.kyma-project.io/access: public** label
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-02-assetgroup.md]: As agreed today, I will keep it in bold.

File: docs/rafter/02-03-asset-and-bucket.md [hunk 36]
```diff
@@ -0,0 +1,20 @@
...
+
+![](./assets/rafter-architecture-2.svg)
+
+1. Create a bucket through a Bucket CR.
+2. The Bucket Controller listens for new Events and acts upon receiving the Bucket CR creation Event.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/02-03-asset-and-bucket.md]: yup, you're rights, probably events here

File: docs/rafter/06-02-assetgroup.md [hunk 37]
```diff
@@ -0,0 +1,112 @@
...
+
+### Status reasons
+
+Processing of an AssetGroup CR can succeed, continue, or fail for one of these reasons:
+
+| Reason | Phase | Description |
+| --------- | ------------- | ----------- |
+| `AssetCreated` | `Pending` | The AssetGroup Controller created the specified asset. |
+| `AssetCreationFailed` | `Failed` | The AssetGroup Controller couldn't create the specified asset due to the provided error. |
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/06-02-assetgroup.md]: I changed it everywhere.

File: docs/rafter/01-01-rafter.md [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+title: Overview
+type: Overview
+---
+
+Technically speaking, Rafter is a solution for storing and managing different types of assets, such as documents, files, images, API specifications, and client-side applications. It uses an external solution, [MinIO](https://min.io/), for storing assets. The whole concept relies on Kubernetes custom resources (CRs) managed by Asset, Bucket, AssetGroup controllers (and their cluster-wide counterparts) grouped under the [Rafter Controller Manager](https://github.com/kyma-project/rafter/blob/master/cmd/manager/README.md). These CRs include:
+
+- Asset CR which manages a single asset or a package of assets
+- Bucket CR which manages buckets in which these assets are stored
+- AssetGroup CR which manages a group of Asset CRs of a specific type
+
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/rafter/01-01-rafter.md]: 🤔

**Comment by kazydek** [comment]
All fixes and a new PR: https://github.com/kyma-project/kyma/pull/6418

**Pull request closed without merging**